[Connector API] Add 8.13 docs (#105456)
elastic/elasticsearch: docs/reference/connector/apis/update-connector-service-type-api.asciidoc

[[update-connector-service-type-api]]
=== Update connector service type API
++++
<titleabbrev>Update connector service type</titleabbrev>
++++

preview::[]

Updates the `service_type` of a connector.

[[update-connector-service-type-api-request]]
==== {api-request-title}

`PUT _connector/<connector_id>/_service_type`

[[update-connector-service-type-api-prereq]]
==== {api-prereq-title}

* To sync data using connectors, it's essential to have the Elastic connectors service running.
* The `connector_id` parameter should reference an existing connector.

> * The `service_type`` must be a valid type as defined by the Connector framework.

Reviewer: ```suggestion
* The `service_type` must be a valid type as defined by the Connector framework.
```

> * When changing a configured connector's `service_type`, reset its configuration as well to ensure compatibility.

Reviewer: ```suggestion
** When you change a connector's `service_type`, you'll also need to reset its configuration to ensure compatibility.
```

[[update-connector-service-type-api-path-params]]
==== {api-path-parms-title}

`<connector_id>`::
(Required, string)

[role="child_attributes"]
[[update-connector-service-type-api-request-body]]
==== {api-request-body-title}

`service_type`::

> (Required, string) A connector service type defined in the https://github.com/elastic/connectors/blob/main/connectors/config.py#L94[Connector framework].

Reviewer: Do we want to link to a specific line of code, seeing as this is prone to change?
